packages/transformers/js/core/src/hoist.rs

```diff
@@ -60,51 +60,266 @@ pub fn hoist(
+  pub imported_symbols: Vec<ImportedSymbol>,
+  /// A vector of the symbols exported from this file, along with their mangled replacement
+  /// identifiers.
+  ///
+  /// For example, if a source file is:
+  ///
+  /// ```skip
+  /// export const x = 1234;
+  /// export function something() {}
+  /// ```
+  ///
+  /// The transformer will replace all usages of `x` and `something` with a mangled generated name.
+  /// For example, the output will look like:
+  ///
+  /// ```skip
+  /// const $abc$export$hashfashdfasdfahsdfh_x = 1234;
+  /// function $abc$export$hashfashdfasdfahsdfh_something() {}
+  /// ```
+  ///
+  ///
+  /// This `exported_symbols` vector will be:
+  ///
+  /// ```skip
+  /// vec![
+  ///     ExportedSymbol {
+  ///         exported: "x",
+  ///         local: "$abc$export$hashfashdfashdfahsdfh_x",
+  ///         ...
+  ///     },
+  ///     ExportedSymbol {
+  ///         exported: "something",
+  ///         local: "$abc$export$hashfashdfashdfahsdfh_something",
+  ///         ...
+  ///     },
+  /// ]
+  /// ```
+  pub exported_symbols: Vec<ExportedSymbol>,
+  /// Symbols re-exported from other modules.
+  ///
+  /// If a symbol is re-exported from another module, parcel will remove delete the export statement
+  /// from the asset.
+  ///
+  /// For example, if an input file is:
+  ///
+  /// ```skip
+  /// export { view as mainView } from './view';
+  /// ```
+  ///
+  /// The output will be
+  /// ```skip
+  /// import 'abc:./view:esm';
+  /// ```
+  ///
+  /// And this vector will contain the information about the re-exported symbol.
+  ///
+  /// On this case, the fields of `ImportedSymbol` will mean different things than they do for
+  /// [`HoistResult::imported_symbols`].
+  ///
+  /// In particular, since there is no mangled name, `local` means the "exported" name rather than
+  /// the mangled name.
+  ///
+  /// On the case above, this field would be:
+  ///
+  /// ```skip
+  /// vec![
+  ///     ImportedSymbol {
+  ///         source: "./view",
+  ///         local: "mainView",
+  ///         imported: "view",
+  ///         ...
+  ///     },
+  /// ]
+  /// ```
+  pub re_exports: Vec<ImportedSymbol>,
+  /// A vector of the 'original local' names of exported symbols that are self-referenced within the
+  /// file they are being exported from.
+  ///
+  /// For example, if a file is:
+  /// ```skip
+  /// exports.foo = 10;
+  /// exports.something = function() {
+  ///     return exports.foo;
+  /// }
+  /// ```
+  ///
+  /// `self_references` will contain the `foo` symbol, un-mangled. Note the output will be mangled:
+  /// ```skip
+  /// var $abc$export$6a5cdcad01c973fa;
+  /// var $abc$export$ce14ccb78c97a7d4;
+  /// $abc$export$6a5cdcad01c973fa = 10;
+  /// $abc$export$ce14ccb78c97a7d4 = function() {
+  ///     return $abc$export$6a5cdcad01c973fa;
+  /// };
+  /// ```
+  pub self_references: HashSet<JsWord>,
+  /// TODO: What is this?
+  pub wrapped_requires: HashSet<String>,
```

Comment:
A non-top-level require, that needs to be wrapped. A top-level require is
```
const/var/let ... = require("something");
```
so the standard CJS import pattern.

Reply:
So is this the mangled variable put to replace the `require(...)` statement?

Reply:
No, that should be handled with the other maps (`imports`).
This is to mark the dependencies with the specifier listed in `wrapped_requires` with `dep.meta.shouldWrap = true`, so that the packager ensure that the require is not executed immediately (as for ES6 `import "...";`).

Reply:
Got it, so this would be the `source` string?

In `require('something')`, it'd be `'something'`?